test/kubernetes/e2e/features/validation/validation_strict_warnings/suite.go

```diff
@@ -152,14 +159,21 @@ func (s *testingSuite) TestInvalidUpstreamMissingPort() {
 	})
 
 	// Upstream is only rejected when the upstream plugin is run when a valid cluster is present
+	// Upstreams no longer report status if they have not been translated at all to avoid conflicting with
+	// other syncers that have translated them, so we can only detect that the objects exist here
 	err = s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, validation.ExampleUpstream, "-n", s.testInstallation.Metadata.InstallNamespace)
 	s.Assert().NoError(err, "can apply valid upstream")
-	s.testInstallation.Assertions.EventuallyResourceStatusMatchesState(
-		func() (resources.InputResource, error) {
+	s.testInstallation.Assertions.EventuallyResourceExists(
+		func() (resources.Resource, error) {
 			return s.testInstallation.ResourceClients.UpstreamClient().Read(s.testInstallation.Metadata.InstallNamespace, validation.ExampleUpstreamName, clients.ReadOpts{Ctx: s.ctx})
 		},
-		core.Status_Accepted,
-		gloo_defaults.GlooReporter,
+	)
+	// we need to make sure Gloo has had a chance to process it
```

Comment:
nit: Does this assert that Gloo had a chance to process it? Wouldn't we want an accepted status to be used to prove that gloo processed it?

Reply:
this is where it starts breaking down. you're right that it doesn't actually assert gloo has processed it, or in reality what we really care about is that it has been added to the internal api snapshot.

Accepted status previously did that, but the problem we are solving is that it was reporting Accepted without doing translation in some cases, this the flicker.

Now that we dont report Accepted naively, we cant use it for a signal that it is in the api snapshot.

All of the tests where i added this are written where the Upstream is applied and exists in the snapshot before a proxy is generated, or really a VirtualService is created.

So this is a cheap way of giving gloo time for the upstream to be created and added to the api snapshot.
We could have just done a sleep as well.

Unfortunately i dont see a good way around this if we have different syncers operating/processing a single config e.g. Upstreams